expandedTree/expandedTree.go

```diff
@@ -36,24 +36,35 @@ func ReadTree(newickFilename string, fastasFilename string) *ETree {
 
 //read in tree from filename
 func ReadNewick(filename string) (*ETree, error) {
-	var line string
+	var singleLineTree string
+	singleLineTree = ReadMultiLineTree(filename)
 
 	file, err := os.Open(filename)
 	if err != nil {
 		return nil, err
 	}
 	defer file.Close()
-	scanner := bufio.NewScanner(file)
 
-	for scanner.Scan() {
-		line = scanner.Text()
-		if !strings.HasPrefix(line, "#") {
-			return parseNewick(line[strings.Index(line, "("):(1 + strings.LastIndex(line, ";"))])
-		}
+	if !strings.HasPrefix(singleLineTree, "#") {
+		return parseNewick(singleLineTree[strings.Index(singleLineTree, "(") : 1+strings.LastIndex(singleLineTree, ";")])
 	}
+
 	return nil, errors.New("Error: tree file is either empty or has no non-comment lines")
 }
 
+func ReadMultiLineTree(input string) string {
```

Comment:
This function looks great!
I think you should consider changing the name or adding a little to the function.  What this function really does is more general that just reading a tree file into a single string (with new lines removed); the function reads *any* file into a string (with line breaks removed).  One option would be to move this into the fileio package and rename it to something like ReadFileToSingleLine.  Another option would be to run the checks and call parseNewick within this function so that it is specific to trees and gives a tree back.

Reply:
I like the idea of moving it to a larger package a lot, I hadn't thought of that, can we get a general consensus on that at update Thursday this week then and then I'll push that change rather than this one?

Reply:
I think you can take a shot at the solution you like better.  You can either:
1) keep it general (any file type to a string of everything in that file with no line breaks) and put it in fileio with a general name, or
2) you can leave it here, but make it specific to reading trees by putting the conversion from string to tree inside this function.
You can ask some other people to review this pull request and get their opinion as well.

Reply:
I'm happy with either solution, but I also quite like the idea of this as a fileio function. I think reading files to single strings might be useful in other contexts.